spring-data-jpa/src/test/java/org/springframework/data/jpa/repository/query/EqlQueryRendererTests.java

```diff
@@ -0,0 +1,991 @@
+				    FROM Employee spouseEmp
+				    WHERE spouseEmp = emp.spouse)
+				""");
+	}
+
+	@Test
+	void allExample() {
+
+		assertQuery("""
+				SELECT emp
+				FROM Employee emp
+				WHERE emp.salary > ALL (SELECT m.salary
+				FROM Manager m
+				WHERE m.department = emp.department)
+				    """);
+	}
+
+	@Test
+	void existsSubSelectExample2() {
+
+		assertQuery("""
+				SELECT DISTINCT emp
+				FROM Employee emp
+				WHERE EXISTS (SELECT spouseEmp
+				    FROM Employee spouseEmp
+				    WHERE spouseEmp = emp.spouse)
+				    """);
+	}
+
+	@Test
+	void subselectNumericComparisonExample1() {
+
+		assertQuery("""
+				SELECT c
+				FROM Customer c
+				WHERE (SELECT AVG(o.price) FROM c.orders o) > 100
+				""");
+	}
+
+	@Test
+	void subselectNumericComparisonExample2() {
+
+		assertQuery("""
+				SELECT goodCustomer
+				FROM Customer goodCustomer
+				WHERE goodCustomer.balanceOwed < (SELECT AVG(c.balanceOwed)/2.0 FROM Customer c)
+				""");
+	}
+
+	@Test
+	void indexExample() {
+
+		assertQuery("""
+				SELECT w.name
+				FROM Course c JOIN c.studentWaitlist w
+				WHERE c.name = 'Calculus'
+				AND INDEX(w) = 0
+				""");
+	}
+
+	/**
+	 * @see #functionInvocationExampleWithCorrection()
+	 */
+	@Test
+	@Disabled(SPEC_FAULT + "FUNCTION calls needs a comparator")
+	void functionInvocationExample_SPEC_BUG() {
+
+		assertQuery("""
+				SELECT c
+				FROM Customer c
+				WHERE FUNCTION('hasGoodCredit', c.balance, c.creditLimit)
+				""");
+	}
+
+	@Test
+	void functionInvocationExampleWithCorrection() {
+
+		assertQuery("""
+				SELECT c
+				FROM Customer c
+				WHERE FUNCTION('hasGoodCredit', c.balance, c.creditLimit) = TRUE
+				""");
+	}
+
+	@Test
+	void updateCaseExample1() {
+
+		assertQuery("""
+				UPDATE Employee e
+				SET e.salary =
+				    CASE WHEN e.rating = 1 THEN e.salary*1.1
+				         WHEN e.rating = 2 THEN e.salary*1.05
+				         ELSE e.salary*1.01
+				    END
+				    """);
+	}
+
+	@Test
+	void updateCaseExample2() {
+
+		assertQuery("""
+				UPDATE Employee e
+				SET e.salary =
+				    CASE e.rating WHEN 1 THEN e.salary*1.1
+				                  WHEN 2 THEN e.salary*1.05
+				                  ELSE e.salary*1.01
+				    END
+				    """);
+	}
+
+	@Test
+	void selectCaseExample1() {
+
+		assertQuery("""
+				SELECT e.name,
+				    CASE TYPE(e) WHEN Exempt THEN 'Exempt'
+				                 WHEN Contractor THEN 'Contractor'
+				                 WHEN Intern THEN 'Intern'
+				                 ELSE 'NonExempt'
+				    END
+				FROM Employee e
+				WHERE e.dept.name = 'Engineering'
+				""");
+	}
+
+	@Test
+	void selectCaseExample2() {
+
+		assertQuery("""
+				SELECT e.name,
+				       f.name,
+				       CONCAT(CASE WHEN f.annualMiles > 50000 THEN 'Platinum '
+				                   WHEN f.annualMiles > 25000 THEN 'Gold '
+				                   ELSE ''
+				              END,
+				       'Frequent Flyer')
+				FROM Employee e JOIN e.frequentFlierPlan f
+				""");
+	}
+
+	@Test
+	void theRest() {
```

Comment:
`theRest<n>` isn't a good naming in particular because it doesn't provide any information. Any chance we can find a better naming that at least gives hints what we're up to?

Reply:
I copied that over from JPQL tests. It was a bit of a shortcut. I'll fix it.